update node version in request to latest

chloe-hubble opened the pull request:
request was using node version 10.13.0 which has long stopped receiving Security Support.

cjmontgom committed:
Update the way we set hubble-network in docker

The build was broken when running/testing this project locally

cjmontgom committed:
Bump node version in docker image, package.json & circle config

We were using an old, unsupported node version which could pintroduce security issues

cjmontgom committed:
Remove 'git add' command from lint-staged config

It's no longer necessary from v10 of lint-staged onwards since it's now automatic, and was outputting a warning to the console

cjmontgom committed:
update isomorphic-unfetch package

we were a major version behind so while updating our node version I thought I may as well

oilyquack reviewed (approved):
Did you also try bumping to version 18? If that's possible without any issue that saves us a job come September when v16 is end of life.

simon360 commented on package.json:
```diff
@@ -38,7 +38,7 @@
   },
   "engines": {
-    "node": ">= 10.13.0",
+    "node": "^16.0.0",
```
I think we'll want to use
```suggestion
    "node": ">= 16.0.0",
```
here.

`^16.0.0` will mean this package can only be installed on Node v16. In other words: it'll raise an error (or warning, at least) if we install it on a project on Node 18.

But `>= 16.0.0` will mark it as working when Node is _at least_ v16, which is closer to what we want.

(I wish JSON had support for comments so we could document the reasoning here!)

chloe-hubble replied:
I'll make the commit message clear and hopefully that will help explain

ghost commented:
> Did you also try bumping to version 18? If that's possible without any issue that saves us a job come September when v16 is end of life.

@simon360 left a comment in the ticket for this which I was going off- "As such, we will want to make sure that we set this field to the earliest version of Node we continue to support. I'd suggest setting it to 16."
But happy to go with the consensus. @oilyquack

oilyquack commented:
> > Did you also try bumping to version 18? If that's possible without any issue that saves us a job come September when v16 is end of life.
> 
> @simon360 left a comment in the ticket for this which I was going off- "As such, we will want to make sure that we set this field to the earliest version of Node we continue to support. I'd suggest setting it to 16." But happy to go with the consensus. @oilyquack

I see, makes sense. Ignore my comment then!

cjmontgom committed:
Set the node version to bigger or equal to 16.0.0

So that any project that uses this has to have a minimum version of Node 16, but won't see any warnings/errors if they have a version higher than 16.
